lib/openstack_api/openstack_query_base.py

```diff
@@ -0,0 +1,48 @@
+from abc import abstractmethod
+from typing import Dict, Callable, Any
+
+from openstack_api.dataclasses import QueryParams
+from openstack_api.openstack_connection import OpenstackConnection
+from openstack_api.openstack_query import OpenstackQuery
+from openstack_api.openstack_wrapper_base import OpenstackWrapperBase
+
+
+class OpenstackQueryBase(OpenstackWrapperBase):

```

Comment:
This should be a plain class
It means we will have to init twice, but it fits the design pattern prefer composition over inheritance which (in the long-term) leads to cleaner code

Else someone changing the WrapperBase may indirectly affect the QueryBase despite them being "unrelated" apart from the connection_cls